lib/hotel.rb

```diff
@@ -0,0 +1,67 @@
+require_relative 'reservation'
+require 'pry'
+
+# class hotel should store and manage the room and reservation lists
+
+class Hotel
+  attr_reader :rooms, :reservations
+  # the ability to create a hotel instance
+  def initialize
+    @rooms = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]# 1-20
+    @reservations = []
+  end
+
+
+  # find an available room for the new reservation
+  # assign the room number to the new reservation
+  def find_available_room
+    if reservations.length == 0
+      room_number = @rooms.sample
+    else
+      # go through @rooms array and sample a room_number
+      # with that room_number call reserevation_overlap?
+      # determine whether the new reservations date range overlaps
+      # with reservations in the reservation array
```

Comment:
Doesn't quite seem like the code below accomplished everything you've outlined here